plugins/inventory/incus.py

```diff
@@ -0,0 +1,220 @@
+  filters:
+    description:
+      - Filter expression as supported by C(incus list).
+    type: list
+    elements: string
+    default: []
+  host_domain:
+    description:
+      - Domain to append to the host FQDN.
+    type: string
+  host_fqdn:
+    description:
+      - Whether to generate a FQDN for the host name.
+      - This will use the INSTANCE.PROJECT.REMOTE syntax.
+    type: bool
+    default: true
+  remotes:
+    description:
+      - The names of the Incus remotes to use (per C(incus remote list)).
+      - Remotes are used to access multiple servers from a single client.
+      - By default the inventory will go over all projects for each remote.
+      - It's possible to specify a specific project using "remote:project".
+    type: list
+    elements: string
+    default: ["local"]
+"""
+
+EXAMPLES = r"""
+---
+# Pull instances from all projects on the local remote.
+plugin: community.general.incus
+
+---
+# Pull running VMs from all projects on the local remote.
+plugin: community.general.incus
+filters:
+  - type=virtual-machine
+  - status=running
+
+---
+# Pull instances from two different remotes
+plugin: community.general.incus
+remotes:
+  - remote-1
+  - remote-2
+
+---
+# Pull instances from two different remotes
+# Limiting the second to the default project
+plugin: community.general.incus
+remotes:
+  - remote-1
+  - remote-2:default
+"""
+
+from ansible.plugins.inventory import BaseInventoryPlugin
+from ansible.utils.display import Display
+from json import loads
+from subprocess import check_output
+
+display = Display()
+
+
+class InventoryModule(BaseInventoryPlugin):
+    """Host inventory parser for Incus."""
+
+    NAME = "community.general.incus"
+
+    def __init__(self):
+        super(InventoryModule, self).__init__()
+
+    def verify_file(self, path):
+        valid = False
+
+        if super(InventoryModule, self).verify_file(path):
+            if path.endswith(("incus.yaml", "incus.yml")):
+                valid = True
+            else:
+                self.display.vvv(
+                    'Skipping due to inventory source not ending in "incus.yaml" nor "incus.yml"'
+                )
+
+        return valid
+
+    def parse(self, inventory, loader, path, cache=True):
+        super(InventoryModule, self).parse(inventory, loader, path)
+
+        self._read_config_data(path)
+
+        # Create top-level "incus" group if missing.
+        self.inventory.add_group("incus")
+
+        for remote in self.get_option("remotes"):
+            # Split the remote name from the project name (if specified).
+            remote_name = ""
+            project_name = ""
+
+            fields = remote.split(":", 1)
+            if len(fields) == 2:
+                remote_name = fields[0]
+                project_name = fields[1]
+            else:
+                remote_name = fields[0]
+
+            # Create the remote-specific group if missing.
+            group_remote = "incus_" + remote_name
```

Comment:
You can use f-strings now:
```suggestion
            group_remote = f"incus_{remote_name}"
```